### PR TITLE
execution-plan-bind: update document for sql bind 

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What is changed, added or deleted? (Required)
Update the execution-plan-bind's document due to the lastest bug-fix and behavior change.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

**Comment by TomShawn** [comment]
@eurekaka @winoros @SunRunAway PTAL

File: reference/performance/execution-plan-bind.md [hunk 1]
```diff
@@ -53,6 +53,10 @@ select * from t where a >    1
 select * from t where a > ？
 ```
 
+通过 `CREATE BINDING` 创建 SQL 绑定，对于每个标准化的 SQL 只能同时存在一个，当对于相同的标准化 SQL 创建多个绑定时，会保留最后一个创建的绑定，之前的所有绑定（创建的和演进出来的）都会被标记为已删除。但是我们仍然允许 session 绑定和 global 绑定共存，它们之间不受这个逻辑影响。
```
Comment (ran-huang) [inline reference/performance/execution-plan-bind.md:56]:
```suggestion
通过 `CREATE BINDING` 创建 SQL 绑定，对于每个标准化的 SQL 只能同时存在一个绑定，当对于相同的标准化 SQL 创建多个绑定时，会保留最后一个创建的绑定，之前的所有绑定（创建的和演进出来的）都会被标记为已删除。但是我们仍然允许 session 绑定和 global 绑定共存，它们之间不受这个逻辑影响。
```

File: reference/performance/execution-plan-bind.md [hunk 2]
```diff
@@ -126,9 +130,17 @@ SHOW [GLOBAL | SESSION] BINDINGS [ShowLikeOrWhere];
 
 开启自动绑定功能后，每隔 `bind-info-lease`（默认值为 `3s`）会遍历一次 Statement Summary 中的历史 SQL 语句，并为至少出现两次的 SQL 语句自动创建绑定。
 
+> **注意：**
+>
+> 由于 TiDB 存在一些内嵌 SQL 保证一些功能的正确性，所以自动创建绑定时会默认屏蔽内嵌 SQL
```
Comment (ran-huang) [inline reference/performance/execution-plan-bind.md:135]:
```suggestion
> 由于 TiDB 存在一些内嵌 SQL 保证一些功能的正确性，所以自动创建绑定时会默认屏蔽内嵌 SQL。
```

File: reference/performance/execution-plan-bind.md [hunk 3]
```diff
@@ -175,3 +187,24 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+    | hint | 说明            |
+    | -------- | ------------- |
+    | memory_quota |  查询过程最多可以使用多少内存 |
+    | use_toja | 优化器是否考虑把子查询转化为 join |
+    | use_cascades | 是否使用 cascades 优化器 |
+    | no_index_merge | 优化器是否考虑将 index merge 作为一个读表选项 |
+    | read_consistent_replica | 是否强制读表时使用 follower read |
+    | max_execution_time | 查询过程最多消耗多少时间 |
+
+4. 我们认为 `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读，由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
```
Comment (ran-huang) [inline reference/performance/execution-plan-bind.md:210]: `无视其所有演进的绑定` 中的 `其` 是指？
```suggestion
4. 我们认为 `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读。由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
```

**Pull request merged**